### ISSUE TITLE
Unable to "unbond" / unbless a production device

### ISSUE THREAD

**Issue by deldrid1** [NasibMansour/imp-central-impt#34]
@NasibMansour - I'm not sure if this is an issue in upstream `impt` or if this was introduced by multiple auth's, but unblessing a production device is not working.

The working request on impCentral looks like this in `curl` form
```bash
curl 'https://api.electricimp.com/v5/devicegroups/b6f010ca-1532-c71f-3963-3c37220a842a/relationships/devices' \ 
-X DELETE \ 
-H 'Origin: https://impcentral.electricimp.com' \ 
-H 'Authorization: [hidden]'  \
-H 'Content-Type: application/vnd.api+json' \
-H 'X-Electricimp-Key: [hidden]' \
-H 'Accept: */*' \
-H 'X-EI-Ide: impcentral.electricimp.com' \
-H 'Referer: https://impcentral.electricimp.com/ide/devices/40000c2a6911315d' \
-H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.142 Safari/537.36' \
-H 'DNT: 1' --data-binary '{"data":[{"id":"40000c2a6911315d","type":"device"}]}' --compressed
```

When ran through `impt`, this is the result (ignore the first request, which is getting the device group ID to put into the second request):
```bash
$ impt device unassign -d 40000c2a6911315d -u "[hidden]" -z debug
Doing the request with options:
{
  "url": "https://api.electricimp.com/v5/devices/40000c2a6911315d",
  "method": "GET",
  "headers": {
    "Content-type": "application/vnd.api+json",
    "Authorization": "[hidden]"
  },
  "json": true,
  "qs": null,
  "body": null,
  "qsStringifyOptions": {
    "arrayFormat": "repeat"
  },
  "forever": true
}

Response code: 200
Response headers: {
  "date": "Thu, 08 Aug 2019 14:37:09 GMT",
  "content-type": "application/vnd.api+json",
  "transfer-encoding": "chunked",
  "connection": "keep-alive",
  "server": "nginx/1.10.3 (Ubuntu)",
  "content-language": "en",
  "x-node": "api02",
  "cache-control": "no-cache, no-store, must-revalidate",
  "pragma": "no-cache",
  "expires": "0",
  "accept": "application/vnd.api+json",
  "x-ratelimit-limit": "50",
 "x-ratelimit-reset": "1",
  "x-ratelimit-remaining": "49",
  "strict-transport-security": "max-age=1209600"
}
Response body: {
  "data": {
    "type": "device",
    "id": "40000c2a6911315d",
    "attributes": {
      "name": null,
      "agent_id": "N5Mzs0TxF_22",
      "agent_url": "https://agent.electricimp.com/N5Mzs0TxF_22",
      "device_online": false,
      "device_state_changed_at": null,
      "agent_running": false,
      "agent_state_changed_at": null,
      "last_enrolled_at": "2018-01-31T13:38:09.196Z",
      "imp_type": "imp004m",
      "ip_address": "151.110.127.62",
      "tags": null,
      "mac_address": "0c:2a:69:11:31:5d",
      "free_memory": null,
      "rssi": null,
      "swversion": "e87ddfb - release-38.17 - Thu Nov 15 09:49:52 2018 - production",
      "plan_id": "eb15d45858a2d454"
    },
    "relationships": {
      "owner": {
        "type": "account",
        "id": "bd81b713-4e56-cb4b-eea9-a0949cae9445"
      },
      "product": {
        "type": "product",
        "id": "adb042d2-9276-a7ce-bac8-2e2f85316278"
      },
      "devicegroup": {
        "type": "production_devicegroup",
        "id": "b6f010ca-1532-c71f-3963-3c37220a842a"
      },
      "deployment": {
        "type": "deployment",
        "id": "f5309fe2-dc40-1165-c3d0-ffef738770f5"
      }
    }
  }
}
Doing the request with options:
{
  "url": "https://api.electricimp.com/v5/devicegroups/b6f010ca-1532-c71f-3963-3c37220a842a/relationships/devices",
  "method": "DELETE",
  "headers": {
    "Content-type": "application/vnd.api+json",
    "Authorization": "[hidden]"
  },
  "json": true,
  "qs": null,
  "body": {
    "data": [
      {
        "type": "device",
        "id": "40000c2a6911315d"
      }
    ]
  },
  "qsStringifyOptions": {
    "arrayFormat": "repeat"
  },
  "forever": true
}

Response code: 403
Response headers: {
  "date": "Thu, 08 Aug 2019 14:37:10 GMT",
  "content-type": "application/vnd.api+json",
  "transfer-encoding": "chunked",
  "connection": "keep-alive",
  "server": "nginx/1.10.3 (Ubuntu)",
  "content-language": "en",
  "x-node": "api01",
  "cache-control": "no-cache, no-store, must-revalidate",
  "pragma": "no-cache",
  "expires": "0",
  "accept": "application/vnd.api+json",
  "x-ratelimit-limit": "50",
  "x-ratelimit-reset": "1",
  "x-ratelimit-remaining": "49",
  "strict-transport-security": "max-age=1209600"
}
Response body: {
  "errors": [
    {
      "code": "PX200",
      "status": "403",
      "title": "Invalid Permission",
      "detail": "Production device removal requires a key with unblessing privilege."
    }
  ]
}
ERROR: HTTP/403: Invalid Permission: Production device removal requires a key with unblessing privilege.
{ Error: Invalid Permission: Production device removal requires a key with unblessing privilege.
    at Function._getError (/Users/AustinEldridge/.nvm/versions/node/v10.15.3/lib/node_modules/imp-central-impt/node_modules/imp-central-api/lib/util/HttpHelper.js:116:28)
    at Request [as _callback] (/Users/AustinEldridge/.nvm/versions/node/v10.15.3/lib/node_modules/imp-central-impt/node_modules/imp-central-api/lib/util/HttpHelper.js:159:40)
    at Request.self.callback (/Users/AustinEldridge/.nvm/versions/node/v10.15.3/lib/node_modules/imp-central-impt/node_modules/request/request.js:185:22)
    at Request.emit (events.js:189:13)
    at Request.<anonymous> (/Users/AustinEldridge/.nvm/versions/node/v10.15.3/lib/node_modules/imp-central-impt/node_modules/request/request.js:1161:10)
    at Request.emit (events.js:189:13)
    at IncomingMessage.<anonymous> (/Users/AustinEldridge/.nvm/versions/node/v10.15.3/lib/node_modules/imp-central-impt/node_modules/request/request.js:1083:12)
    at Object.onceWrapper (events.js:277:13)
    at IncomingMessage.emit (events.js:194:15)
    at endReadableNT (_stream_readable.js:1125:12) _statusCode: 403, _body: { errors: [ [Object] ] } }
Error: Invalid Permission: Production device removal requires a key with unblessing privilege.
IMPT COMMAND FAILS
```

It looks like the thing that is missing is the `X-Electricimp-Key` header, which contains the `unbond password` as described [here](http://apidoc.electricimp.com/#tag/Devices/paths/~1devicegroups~1{devicegroup_id}~1relationships~1devices/delete).  Can we get this fixed?

Only other thing to note is that in `-z debug` mode, the unbless key should propbably be hidden from output, just like the `Authorization` header.

**Comment by NasibMansour** [comment]
@deldrid1 should be fixed now, it looks like a silly mistake on imp's side.
You'll need to install the latest version of `impt` from branch `feature/add-multiple-test-groups`

**Comment by NasibMansour** [comment]
Fixed by 7b7ab36825b703dc3e713cb5d45351996693efd3 in #39
